homeassistant/components/azure_devops/__init__.py

```diff
@@ -1,80 +1,104 @@
-                    "Could not authorize with Azure DevOps. You may need to update your token"
-                )
-        await client.get_project(entry.data[CONF_ORG], entry.data[CONF_PROJECT])
-    except aiohttp.ClientError as exception:
-        _LOGGER.warning(exception)
-        raise ConfigEntryNotReady from exception
-
-    instance_key = f"{DOMAIN}_{entry.data[CONF_ORG]}_{entry.data[CONF_PROJECT]}"
-    hass.data.setdefault(instance_key, {})[DATA_AZURE_DEVOPS_CLIENT] = client
+    client: DevOpsClient = DevOpsClient()
+
+    async def async_update_data() -> tuple[
+        DevOpsClient, DevOpsProject, list[DevOpsBuild]
+    ]:
+        """Fetch data from Azure DevOps."""
+
+        try:
+            if entry.data[CONF_PAT] is not None:
+                await client.authorize(entry.data[CONF_PAT], entry.data[CONF_ORG])
+                if not client.authorized:
+                    raise ConfigEntryAuthFailed(
+                        "Could not authorize with Azure DevOps. You may need to update your token"
+                    )
+            project: DevOpsProject = await client.get_project(
+                entry.data[CONF_ORG], entry.data[CONF_PROJECT]
+            )
+            builds: list[DevOpsBuild] = await client.get_builds(
+                entry.data[CONF_ORG], entry.data[CONF_PROJECT], BUILDS_QUERY
+            )
+            return client, project, builds
+        except (aiohttp.ClientError, aiohttp.ClientError) as exception:
+            raise UpdateFailed from exception
+
+    coordinator = DataUpdateCoordinator(
+        hass,
+        _LOGGER,
+        # Name of the data. For logging purposes.
+        name=f"{DOMAIN}_coordinator",
+        update_method=async_update_data,
+        # Polling interval. Will only be polled if there are subscribers.
+        update_interval=timedelta(seconds=300),
+    )
+
+    hass.data.setdefault(DOMAIN, {})
+    hass.data[DOMAIN][entry.entry_id] = coordinator
+
+    # Fetch initial data so we have data when entities subscribe
+    await coordinator.async_config_entry_first_refresh()
 
-    # Setup components
     hass.config_entries.async_setup_platforms(entry, PLATFORMS)
 
     return True
 
 
 async def async_unload_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
     """Unload Azure DevOps config entry."""
-    del hass.data[f"{DOMAIN}_{entry.data[CONF_ORG]}_{entry.data[CONF_PROJECT]}"]
+    unload_ok = await hass.config_entries.async_unload_platforms(entry, PLATFORMS)
+    if unload_ok:
+        del hass.data[DOMAIN][entry.entry_id]
+    return unload_ok
 
-    return await hass.config_entries.async_unload_platforms(entry, PLATFORMS)
 
-
-class AzureDevOpsEntity(Entity):
+class AzureDevOpsEntity(CoordinatorEntity):
     """Defines a base Azure DevOps entity."""
 
-    def __init__(self, organization: str, project: str, name: str, icon: str) -> None:
+    def __init__(
+        self,
+        coordinator: DataUpdateCoordinator,
+        organization: str,
```

Comment:
`organization` is a part of your entity description, you should use that instead